internal/authentication/tokengetter.go

```diff
@@ -19,6 +21,14 @@ type TokenGetter struct {
 	mu                 sync.RWMutex
 }
 
+type ServiceAccountNotFoundError struct {
+	ServiceAccountName string
+}
+
+func (e *ServiceAccountNotFoundError) Error() string {
+	return fmt.Sprintf("Unable to authenticate with Kubernetes cluster using ServiceAccount \"%s\": ServiceAccount \"%s\" not found.", e.ServiceAccountName, e.ServiceAccountName)
+}
```

Comment:
```suggestion
// ServiceAccountNotFoundError represents an error when a specified ServiceAccount is not found.
type ServiceAccountNotFoundError struct {
	ServiceAccountName string // The name of the missing ServiceAccount.
}

// Error implements the error interface for ServiceAccountNotFoundError.
func (e *ServiceAccountNotFoundError) Error() string {
	return fmt.Sprintf("ServiceAccount \"%s\" not found: Unable to authenticate with the Kubernetes cluster.", e.ServiceAccountName)
}
```

Reply:
Hey! Just a quick note:

Anything we write with a capital letter at the front, like a function or struct name, is exported—meaning it can be accessed by other packages. That also means we should add proper documentation for those with // comments on top. This makes the code easier to understand for anyone using it  😊

Reply:
sure